src/main/scala/cognite/spark/v1/DefaultSource.scala

```diff
@@ -85,13 +94,116 @@ class DefaultSource
+        Constants.SparkDatasourceVersion,
+        getFunctionsAuth(config),
+        config.baseUrl,
+        apiVersion = Some("playground"))
+      call <- client
+        .functionCalls(functionId)
+        .callFunction(Json.fromJsonObject(JsonObject(("data", jsonData))))
+      res <- getFunctionResult(client, functionId, call)
+    } yield res.toString
+  }
+
+  private def callFunctionByExternalId(config: RelationConfig, externalId: String, data: String) = {
+    implicit val backend: SttpBackend[IO, Nothing] = CdpConnector.retryingSttpBackend(config.maxRetries)
+    val funcIO = for {
+      client <- GenericClient.forAuth[IO](
+        Constants.SparkDatasourceVersion,
+        getFunctionsAuth(config),
+        config.baseUrl,
+        apiVersion = Some("playground"))
+      func <- client.functions.retrieveByExternalId(externalId)
+    } yield func
+
+    val func = funcIO.unsafeRunSync()
+    callFunctionAndGetResult(config, func.id.get, data)
+  }
+
+  private def callFunctionByName(config: RelationConfig, name: String, data: String) = {
+    implicit val backend: SttpBackend[IO, Nothing] = CdpConnector.retryingSttpBackend(config.maxRetries)
+    val functionsIO = for {
+      client <- GenericClient.forAuth[IO](
+        Constants.SparkDatasourceVersion,
+        getFunctionsAuth(config),
+        config.baseUrl,
+        apiVersion = Some("playground"))
+      functions <- client.functions.read()
+    } yield functions
+
+    val functionOption = functionsIO.unsafeRunSync().items.filter(func => func.name == name).headOption
+    functionOption match {
+      case Some(value) => callFunctionAndGetResult(config, value.id.get, data)
+      case None => IO.pure("")
+    }
+  }
+
+  private def getFunctionsAuth(config: RelationConfig) =
+    config.functionsAuth match {
+      case Some(value) => value
+      case None => config.auth
+    }
+
+  private def initializeFunctionUdfs(config: RelationConfig, context: SQLContext): Unit = {
+    context.sparkSession.udf.register(
+      "cdf_function",
+      (functionId: Long, data: String) =>
+        callFunctionAndGetResult(config, functionId, data).unsafeRunSync()
+    )
+    context.sparkSession.udf.register(
+      "cdf_function_external_id",
+      (externalId: String, data: String) =>
+        callFunctionByExternalId(config, externalId, data).unsafeRunSync()
+    )
+    context.sparkSession.udf.register(
+      "cdf_function_name",
+      (name: String, data: String) => callFunctionByName(config, name, data).unsafeRunSync()
+    )
+  }
+
   // scalastyle:off cyclomatic.complexity method.length
   override def createRelation(
       sqlContext: SQLContext,
       parameters: Map[String, String],
       schema: StructType): BaseRelation = {
+
     val resourceType = parameters.getOrElse("type", sys.error("Resource type must be specified"))
     val config = parseRelationConfig(parameters, sqlContext)
+    initializeFunctionUdfs(config, sqlContext)
```

Comment:
Maybe we should do this separately and make it opt-in, we're registering those UDFs for every user of the data source like this. A separate cognite.spark.v1.udfs package would make sense, we also have some in Jetfire that we can/should share as well. Anyways, we can make that change in a subsequent PR.